### DISCUSSION TITLE
papis seems reluctant to download and its hard to fix once broken

### DISCUSSION THREAD

**Discussion by ExpandingMan** [papis/papis#809] [General]
In the examples I've tried (almost exclusively arxiv), papis will only download a document when I explicitly use `--from`, even though it usually finds the metadata just fine in any case.  This is pretty frustrating, because once I've added a paper but it's missing its pdf, there's no trivial way to restore it.  As far as I can tell, your only option is to manually download the file, then update the metadata either directly in the `yml` or through the command line.

- Am I missing something here?  Perhaps it's a bug that it's not always downloading?  I have `add-download-files = True`.  Again, it *does* work if I have an explicit `--from`.
- It would be really nice if, regardless of anything else, there were a really dead-simple way to fetch the PDF for a file with known metadata.

**Comment by alexfikl** [comment]
Just tried this with the following command and it seems to work:
```
papis -l debug add --download-files 'https://arxiv.org/abs/2312.05606'
```

What version of papis are you using? That's on the current git, so not sure if it still works properly on older versions, since we made some changes around the downloading of files since the last release.

**Comment by ExpandingMan** [comment]
I'm using the AUR version from the package `papis`, which I just realized is declared "orphaned", though it is supposedly latest tagged version (`0.13`).  The `papis-git` package goes through a ridiculously long build process, so I'd rather not use it.  I might need to work out and poetry install or something like that.

**Reply by alexfikl** [reply]
Yeah, "orphaned" just means that it doesn't currently have a maintainer, but that's a fairly recent change.
I'm not sure why the `papis-git` version would have a longer build process though. From looking at them, the main difference seems to be that it builds the manpages.

Any way, version 0.13 does not have the `add-download-files` configuration option, so that won't do anything. There should be a `--force-download` flag (with no accompanying configuration setting), but I'm not sure that worked correctly.

**Comment by ExpandingMan** [comment]
`papis-git` was running tests for some reason, a lot of AUR packages do this in cases where I feel they shouldn't.

Anyway, it sounds like `0.14` is expected relatively soon.  In the meantime I can mitigate this issue by being careful.  I'll be happy to check back in and figure out install options when `0.14` is out.

Thanks!

**Reply by alexfikl** [reply]
If you use `makepkg` directly, you can call it with `makepkg --nocheck` to skip the tests. Not sure about the various AUR helpers, but they should have some way to skip them too.

> Anyway, it sounds like 0.14 is expected relatively soon. In the meantime I can mitigate this issue by being careful. I'll be happy to check back in and figure out install options when 0.14 is out.

Fingers crossed! At least we're working towards it :grin:
